### ISSUE TITLE
CLI option --override-genesis-config not working as it used to

### ISSUE THREAD

**Issue by lucassaldanha** [hyperledger/besu#7439]
### Description

I noticed a side effect of a change in Besu that has made it incompatible with Teku acceptance tests. The TLDR is that Besu is not respecting the parameters we use with the option `--override-genesis-config`.

For example, in Teku acceptance tests, we calculate the genesis time of the network and use the override to set `shanghaiTime`.

This problem seems to have started here:  https://github.com/hyperledger/besu/pull/7012, unfortunately, I started looking into it but it was taking too long and I had other things that I had to do. The gist of the issue is how we are not propagating `Map<String, String> genesisConfigOverrides` down to `ConsensusScheduleBesuControllerBuilder` how we used to.

The result is that some values are being correctly overwritten, while some others are not (it really depends on what object is consuming the correct map of overrides).

### Acceptance Criteria
* `--override-genesis-config` values being correctly applied, as if the values were part of the original genesis file.

### Steps to Reproduce (Bug)
1. Start Besu with `--override-genesis-config="shanghaiTime=123"`

**Expected behavior:** `shanghaiTime` correctly applied on protocol schedule. 

**Actual behavior:** `shanghaiTime` is not correctly applied on protocol schedule.

**Frequency:** Always

### Logs (if a bug)
N/A

### Versions (Add all that apply)
All versions post https://github.com/hyperledger/besu/commit/690a2ea81e2bf62508ff16675cd7dbaeb006fa83

**Comment by macfarla** [comment]
besu 24.3.3 was the last release that had this working properly.

`besu-24.3.3 ./bin/besu --override-genesis-config="shanghaiTime=123"`

```
2024-08-13 14:06:13.595+10:00 | main | ERROR | Besu | Failed to start Besu
picocli.CommandLine$ExecutionException: Genesis Config Error: 'Shanghai' is scheduled for milestone 123 but it must be on or after milestone 15050000.
```

**Comment by macfarla** [comment]
started looking into this - added 2 tests that fail https://github.com/hyperledger/besu/compare/main...macfarla:override-gen?expand=1

looks like the problem is that we used to pass the overrides to updateNetworkConfig() each time but now we rely on GenesisConfigFile.getConfigOptions() - but GenesisConfigFile does not keep a reference to the overrides, they need to be passed in eg GenesisConfigFile.getConfigOptions(Map overrides)

So think the solution is to refactor GenesisConfigFile to also keep the overrides (but they are technically not part of the _file_), OR use GenesisConfigOptions when creating BesuControllerBuilder, instead of GenesisConfigFile
